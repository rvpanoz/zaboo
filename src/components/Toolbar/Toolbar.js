import React from "react";
import { useDispatch } from "react-redux";
import Toolbar from "@material-ui/core/Toolbar";
import Button from "@material-ui/core/Button";
import { evalCode } from "actions/system/actions";

const AppToolbar = () => {
  const dispatch = useDispatch();
  const onClick = () => dispatch(evalCode());

  return (
    <div>
      <Toolbar>
        <Button color="inherit" onClick={onClick}>
          Visualize
        </Button>
      </Toolbar>
    </div>
  );
};

export default AppToolbar;
