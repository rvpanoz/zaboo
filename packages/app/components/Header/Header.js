import React from "react";
import { useDispatch } from "react-redux";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";
import MenuIcon from "@material-ui/icons/Menu";
import SearchIcon from "@material-ui/icons/Search";
import clsx from "clsx";
import styles from "./styles";
import { signout } from "../../actions/user/actions";

const useStyles = makeStyles(styles);

const Header = ({ sidebarOpen, toggleDrawer }) => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const signoutUser = () => dispatch(signout());

  return (
    <div className={classes.root}>
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: sidebarOpen
        })}
      >
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="open drawer"
            onClick={toggleDrawer}
          >
            <MenuIcon />
          </IconButton>
          <Typography className={classes.title} variant="h6" noWrap>
            zaboo
          </Typography>
          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput
              }}
              inputProps={{ "aria-label": "search" }}
            />
          </div>
          <Button color="inherit" onClick={signoutUser}>
            Sign out
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default Header;
