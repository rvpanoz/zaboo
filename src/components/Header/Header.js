import React from "react";
import { func, bool } from "prop-types";
import { useDispatch } from "react-redux";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";
import MenuIcon from "@material-ui/icons/Menu";
import clsx from "clsx";
import { signout } from "actions/user/actions";
import styles from "./styles";

const useStyles = makeStyles(styles);

const Header = ({ sidebarOpen, toggleDrawer }) => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const signoutUser = () => dispatch(signout());

  return (
    <div className={classes.root}>
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: sidebarOpen
        })}
      >
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="open drawer"
            onClick={toggleDrawer}
          >
            <MenuIcon />
          </IconButton>
          <Typography className={classes.title} variant="h6" noWrap>
            zaboo
          </Typography>
          <Button color="inherit" onClick={signoutUser}>
            Sign out
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
};

Header.propTypes = {
  sidebarOpen: bool.isRequired,
  toggleDrawer: func.isRequired
};

export default Header;
