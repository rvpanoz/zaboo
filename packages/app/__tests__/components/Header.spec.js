import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Header from "../../components/Header";
import Toolbar from "@material-ui/core/Toolbar";

describe("Header component", () => {
  describe("UI tests", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Header />);
    });

    it.skip("renders a <Toolbar /> component", () => {
      const result = wrapper.find(Toolbar).length;
      expect(result).toBe(1);
    });
  });

  describe("Snapshot tests", () => {
    it.skip("renders correctly using renderer", () => {
      const tree = renderer.create(<Header />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
