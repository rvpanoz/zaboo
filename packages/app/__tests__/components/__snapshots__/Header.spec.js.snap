// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Snapshot tests renders correctly using renderer 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-2 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-6 MuiTypography-h6 MuiTypography-noWrap"
      >
        japp
      </h6>
      <div
        className="makeStyles-search-7"
      >
        <div
          className="makeStyles-searchIcon-8"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="MuiInputBase-root makeStyles-inputRoot-9"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            className="MuiInputBase-input makeStyles-inputInput-10"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Searchâ€¦"
            type="text"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
